istioctl/cmd/debug.go

```diff
@@ -110,6 +113,46 @@ By default it will use the default serviceAccount from (istio-system) namespace
 					TypeUrl: TypeDebug,
 				}
 			}
+			if centralOpts.Xds == "" {
+				svc, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), istiodServiceName, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("please specify %q as %v", "--xds-address", err)
+				}
+				namespace, selector, err := polymorphichelpers.SelectorsForObject(svc)
+				if err != nil {
+					return fmt.Errorf("please specify %q as we cannot attach to %T: %v", "--xds-address", svc, err)
+				}
+
+				options := metav1.ListOptions{LabelSelector: selector.String()}
+
+				podList, err := kubeClient.CoreV1().Pods(namespace).List(context.TODO(), options)
+				if err != nil {
+					return fmt.Errorf("please specify %q as %v", "--xds-address", err)
+				}
+				//  select a pod randomly to simulate current debug behavior
+				pod := podList.Items[0]
```

Comment:
This will always select first pod and not random pod?

Reply:
suppose the `List()` will not have fixed order for returned Items.